core/src/test/java/nl/nn/adapterframework/util/MiscTest.java

```diff
@@ -1,43 +1,610 @@
+        assertEquals(os.toString(), "inside the lebron file");
+    }
+
+    /**
+     * Method: streamToStream(InputStream input, OutputStream output)
+     */
+    @Test
+    public void testStreamToStreamForInputOutput() throws Exception {
+        String test = "test";
+        ByteArrayInputStream bais = new ByteArrayInputStream(test.getBytes());
+        OutputStream baos = new ByteArrayOutputStream();
+        Misc.streamToStream(bais, baos);
+        assertEquals(baos.toString(), "test");
+    }
+
+    /**
+     * Method: streamToStream(InputStream input, OutputStream output, boolean closeInput)
+     */
+    @Test
+    public void testStreamToStreamForInputOutputCloseInput() throws Exception {
+        String test = "test";
+        ByteArrayInputStream bais = new ByteArrayInputStream(test.getBytes());
+        OutputStream baos = new ByteArrayOutputStream();
+        Misc.streamToStream(bais, baos, true);
+        assertEquals(baos.toString(), "test");
+    }
+
+    /**
+     * Method: streamToFile(InputStream inputStream, File file)
+     */
+    @Test
+    public void testStreamToFile() throws Exception {
+        String test = "test";
+        ByteArrayInputStream bais = new ByteArrayInputStream(test.getBytes());
+        Misc.streamToFile(bais, file);
+
+        // to read from the file
+        InputStream is = new FileInputStream(file);
+        BufferedReader buf = new BufferedReader(new InputStreamReader(is));
+
+        String line = buf.readLine();
+        StringBuilder sb = new StringBuilder();
+
+        while (line != null) {
+            sb.append(line).append("\n");
+            line = buf.readLine();
+        }
+
+        String fileAsString = sb.toString();
+        assertEquals(fileAsString, "test\n");
+    }
+
+    /**
+     * Method: streamToBytes(InputStream inputStream)
+     */
+    @Test
+    public void testStreamToBytes() throws Exception {
+        String test = "test";
+        ByteArrayInputStream bais = new ByteArrayInputStream(test.getBytes());
+        byte[] arr = Misc.streamToBytes(bais);
+        assertEquals(new String(arr, StandardCharsets.UTF_8), "test");
+    }
+
+    /**
+     * Method: readerToWriter(Reader reader, Writer writer)
+     */
+    @Test
+    public void testReaderToWriterForReaderWriter() throws Exception {
+        Reader reader = new StringReader("test");
+        Writer writer = new StringWriter();
+        Misc.readerToWriter(reader, writer);
+        assertEquals(writer.toString(), "test");
+    }
+
+
+    /**
+     * Method: fileToString(String fileName)
+     */
+    @Test
+    public void testFileToStringFileName() throws Exception {
+        //Misc.resourceToString()
+        assertEquals(Misc.fileToString(file.getName()), "inside the lebron file");
+    }
+    @Test
+    public void testFileToStringFileNameEndLine() throws Exception {
+        //Misc.resourceToString()
+        assertEquals(Misc.fileToString(file.getName(), " the end"), "inside the lebron file");
```

Comment:
You're not testing the `endOfLineString` string here?

Reply:
I see you're testing it in another test, maybe to not confuse things leave it blank or `null` instead :)

Reply:
I removed this test. It was not testing anything unique.